nltk/twitter/twitterclient.py

```diff
@@ -0,0 +1,457 @@
+                colditzjb commented on 9 Dec 2014
+            '''
+            try:
+                if track == '' and follow == '' and locations == None:
+                    msg = "Please supply a value for 'track', 'follow' or 'locations'."
+                    raise ValueError(msg)
+                self.statuses.filter(track=track, follow=follow, lang=lang,
+                                     locations=locations)
+            except requests.exceptions.ChunkedEncodingError as e:
+                if e is not None:
+                    print("Error (stream will continue): {0}".format(e))
+                continue
+
+
+
+
+class Query(Twython):
+    """
+    Retrieve data from the Twitter REST API.
+    """
+    def __init__(self, app_key, app_secret, oauth_token,
+                 oauth_token_secret):
+        self.handler = None
+        self.do_continue = True
+        Twython.__init__(self, app_key, app_secret, oauth_token, oauth_token_secret)
+
+    def register(self, handler):
+        """
+        Register a method for handling tweets.
+
+        :param TweetHandlerI handler: method for viewing or writing tweets to a file.
+        """
+        self.handler = handler
+
+    def _lookup(self, infile, verbose=True):
+        """
+        :param infile: name of a file consisting of Tweet IDs, one to a line
+        :return: iterable of Tweet objects
+        """
+        with open(infile) as f:
+            ids = [line.rstrip() for line in f]
+        if verbose:
+            print("Counted {0} Tweet IDs in {1}.".format(len(ids), infile))
+
+        id_chunks = [ids[i:i+100] for i in range(0, len(ids), 100)]
+        """
+        The Twitter endpoint takes lists of up to 100 ids, so we chunk
+        the ids.
+        """
+
+        listoflists = [self.post('statuses/lookup', {'id': chunk}) for chunk
+                       in id_chunks]
+        return itertools.chain.from_iterable(listoflists)
+
+
+    def lookup(self, infile, outfile, verbose=True):
+        """
+        Given a file containing a list of tweetIDs, fetch the corresponding
+        Tweets (if they haven't been deleted) and dump them in a file.
+
+        :param str infile: Name of a file consisting of tweetIDs, one to a line
+        :param str outfile: Name of file where JSON serialisations of fully\
+        expanded Tweets will be written.
+        """
+        tweets = self._lookup(infile, verbose=verbose)
+        count = 0
+
+        if os.path.isfile(outfile):
+            os.remove(outfile)
+
+        with open(outfile, 'a') as f:
+            for data in tweets:
+                json.dump(data, f)
+                f.write("\n")
+                count += 1
+
+        if verbose:
+            print("""Written {0} Tweets to file {1} of length {2}
+            bytes""".format(count, outfile, os.path.getsize(outfile)))
+
+    def search_tweets(self, keywords, count=100, lang='en', retries_after_twython_exception=0):
```

Comment:
For a few reasons, this would be nicer as a generator. If one lands up in that 15-minute sleep statement, and decides they'd rather look at what they've got so far, they have the choice of waiting, or killing the operation and failing to re-request it.

It also allows the user to determine `count` dynamically. But keeping the `count` parameter means we don't force users to know `itertools.islice`.